adapter/integration-test/src/adapter-datasource.test.js

```diff
@@ -1,77 +1,125 @@
 const request = require('supertest')
 
 const {
-  ADAPTER_URL
+  ADAPTER_URL,
+  AMQP_URL
 } = require('./env')
 const { waitForServicesToBeReady } = require('./waitForServices')
+const {
+  connectAmqp,
+  consumeAmqpMsg
+} = require('./testHelper')
+
+const AMQP_EXCHANGE = 'ods_global'
+const AMQP_QUEUE = 'adapter-it-ds'
+const CONFIG_TOPIC = 'datasource.config.*'
+const CONFIG_CREATED_TOPIC = 'datasource.config.created'
+const CONFIG_DELETED_TOPIC = 'datasource.config.deleted'
+const CONFIG_UPDATED_TOPIC = 'datasource.config.updated'
 
 const TIMEOUT = 10000
+const publishedEvents = new Map() // routing key -> received msgs
+let amqpConnection
 
 describe('Datasource Configuration', () => {
   beforeAll(async () => {
     await waitForServicesToBeReady()
+
+    amqpConnection = await connectAmqp(AMQP_URL)
+
+    await consumeAmqpMsg(amqpConnection, AMQP_EXCHANGE, CONFIG_TOPIC, AMQP_QUEUE, publishedEvents)
   }, 60000)
 
+  afterAll(async () => {
+    await request(ADAPTER_URL)
+      .delete('/configs')
+      .send()
+    if (amqpConnection) {
+      await amqpConnection.close()
+    }
+  }, TIMEOUT)
+
   test('Should respond with stored datasource configs', async () => {
     const response = await request(ADAPTER_URL).get('/datasources')
     expect(response.status).toEqual(200)
     expect(response.type).toEqual('application/json')
 
-    expect(response.body).toEqual([])
+    expect(response.body).toBeInstanceOf(Array)
```

Comment:
Well, that test is non-saying now.
We should either test that a created one is contained, or that it is empty if no data sources are present. I'd prefer the first option!

Reply:
I changed this because I did not want this test to make assumptions about the initial adapter state. Specifically, I wanted to enable running the test locally without having to delete docker volumes first. 
However, maybe, having a stronger test is more important. What do you think?

Reply:
Can we add it to the create test? Like we create one, then use the get and check if it is contained in all configs?

Reply:
Yes, good idea, I will add that